### PR TITLE
Feature: adding list of all tools in marketplace

### PR DESCRIPTION
**Pull request by MedVAL-na**
Task: "Add list of tools to /marketplace"

Done:

- created editor tool repository
- created editor tool domain service
- added new application service for editor tools

### CODE REVIEW

File: src/application/router/routes.ts [hunk 1]
```diff
@@ -55,6 +56,10 @@ const routes: RouteRecordRaw[] = [
       id: String(route.params.id),
     }),
   },
+  {
+    path: `/marketplace/`,
```
Comment (neSpecc) [inline src/application/router/routes.ts:60]:
```suggestion
    path: `/marketplace`,
```

File: src/infrastructure/storage/user.ts [hunk 2]
```diff
@@ -50,6 +55,22 @@ export class UserStore extends SubscribableStore<UserStoreData> {
...
+  public getAllEditorTools(): EditorTool[] {
+    return this.data.allEditorTools;
+  }
+
+  /**
+   * Set all editor tools that are used in notes creation
+   *
+   * @param editorTools - editor plugins
+   */
+  public setAllEditorTools(editorTools: EditorTool[]): void {
```
Comment (neSpecc) [inline src/infrastructure/storage/user.ts:73]: it is not related to user storage and app state. Just create the new application service, domain service, repo

File: src/infrastructure/user.repository.ts [hunk 3]
```diff
@@ -64,10 +64,25 @@ export default class UserRepository extends Repository<UserStore, UserStoreData>
...
       toolId: id,
     });
 
     console.log('Add tool response', response);
   }
+  /**
+   * Load tools and set it
+   */
+  public async loadAllEditorTools(): Promise<void> {
+    const response = await this.transport.get<{ data: EditorTool[] }>('/user/editor-tools');
```
Comment (neSpecc) [inline src/infrastructure/user.repository.ts:77]: why user?

File: src/infrastructure/user.repository.ts [hunk 4]
```diff
@@ -64,10 +64,25 @@ export default class UserRepository extends Repository<UserStore, UserStoreData>
...
+
+    this.store.setAllEditorTools(response.data);
```
Comment (neSpecc) [inline src/infrastructure/user.repository.ts:79]: Is there any reason to store marketplace tools in memory?

File: src/application/services/useAppState.ts [hunk 5]
```diff
@@ -36,11 +37,11 @@ export const useAppState = createSharedComposable((): UseAppStateComposable => {
...
    * Subscribe to user changes in the App State
    */
-  AppStateController.user((prop: 'user' | 'editorTools', value: User | EditorTool[] | null) => {
+  AppStateController.user((prop: 'user' | 'userEditorTools' | 'allEditorTools', value: User | EditorTool[] | null) => {
     if (prop === 'user') {
       user.value = value as User;
     }
-    if (prop === 'editorTools') {
+    if (prop === 'userEditorTools') {
       userEditorTools.value = value as EditorTool[];
```
Comment (neSpecc) [inline src/application/services/useAppState.ts:45]: why do we need this change?

File: src/application/services/useEditorTool.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+/**
+ * Application service for working with the Editor Tools
+ */
+export default function (): UseEditorToolComposable {
+  /**
+   *  All editor tools
+   */
+  const allEditorTools = ref<EditorTool[]>([]);
```
Comment (neSpecc) [inline src/application/services/useEditorTool.ts:22]:
```suggestion
  const tools = ref<EditorTool[]>([]);
```

File: src/application/services/useEditorTool.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+  /**
+   * Get list of all tools
+   */
+  async (): Promise<void> => {
```
Comment (neSpecc) [inline src/application/services/useEditorTool.ts:27]: unclear constrict. Why do we need it ?

File: src/domain/editorTool.service.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import type EditorToolRepository from '@/domain/editorTool.repository.interface';
+import type EditorTool  from './entities/EditorTool';
+
+/**
+ * Business logic for User
```
Comment (neSpecc) [inline src/domain/editorTool.service.ts:5]:
```suggestion
 * Business logic for working with Marketplace
```

File: src/domain/editorTool.service.ts
Comment (neSpecc) [inline src/domain/editorTool.service.ts:1]: lets call it "marketplace.service"

File: src/infrastructure/editorTool.repository.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import type EditorToolRepositoryInterface from '@/domain/editorTool.repository.interface';
+import type NotesApiTransport from './transport/notes-api';
+import type EditorTool from '@/domain/entities/EditorTool';
+
+/**
+ * Facade for the user data
+ */
+export default class EditorToolRepository implements EditorToolRepositoryInterface {
```
Comment (neSpecc) [inline src/infrastructure/editorTool.repository.ts:8]: Let's use "marketpalce" term instead

File: src/infrastructure/editorTool.repository.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+   * @param notesApiTransport - notes api transport instancea
+   */
+  constructor(notesApiTransport: NotesApiTransport) {
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Returns all editor tools
+   */
+  public async getAllEditorTool(): Promise<EditorTool[]> {
```
Comment (neSpecc) [inline src/infrastructure/editorTool.repository.ts:26]:
```suggestion
  public async getToolsAvailable(): Promise<EditorTool[]> {
```

File: src/infrastructure/editorTool.repository.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Get tool data from API
+     */
+    const response = await this.transport.get<{ data: EditorTool[] }>('/editor-tools/all');
+
+    return (response.data);
```
Comment (neSpecc) [inline src/infrastructure/editorTool.repository.ts:32]: why do we need "()" ?

File: src/presentation/pages/Marketplace.vue [hunk 12]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+    <h1>Marketplace</h1>
+    <h2>Tools</h2>
```
Comment (neSpecc) [inline src/presentation/pages/Marketplace.vue:4]: use i18n please for all UI texts

File: src/application/services/useAppState.ts (hunk 5 above)
Comment (MedVAL-na) [reply src/application/services/useAppState.ts]: Because this is only for user tools. This naming was in @GeekaN2 PR, and it is used specifically for downloading user tools in /settings page

Comment (neSpecc) [reply src/application/services/useAppState.ts]: it is already a subscription to the "user" state, all props here are related to user. Anyway, this change is not related to Marketplace, so it's better to clear all unrelated changes

File: src/application/services/useEditorTool.ts (hunk 7 above)
Comment (MedVAL-na) [reply src/application/services/useEditorTool.ts:27]: For referring to service's method which will load list of tools

File: src/application/services/useAppState.ts
Comment (neSpecc) [inline src/application/services/useAppState.ts:1]: looks like unrelated changes

File: src/application/services/useEditorTool.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+  /**
+   * All editor tools that are used in notes creation
+   */
+  tools: Ref<EditorTool[]>
+}
+
+/**
+ * Application service for working with the Editor Tools
+ */
+export default function (): UseEditorToolComposable {
```
Comment (neSpecc) [inline src/application/services/useEditorTool.ts:18]: lets rename it to useMarketplace

File: src/application/services/useEditorTool.ts (hunk 7 above)
Comment (neSpecc) [reply src/application/services/useEditorTool.ts:27]: try moving it to `onMounted` instead

File: src/presentation/components/editor/Editor.vue
Comment (neSpecc) [inline src/presentation/components/editor/Editor.vue:1]: unrelated changes

**Review by neSpecc — COMMENTED** [review]
Almost done 👍, few little changes left

File: src/presentation/components/editor/Editor.vue
Comment (MedVAL-na) [reply src/presentation/components/editor/Editor.vue:1]: This changes is nedeed because we renamed `userEditorTools ` to `editorTools` in useAppState.

File: src/application/services/useAppState.ts
Comment (MedVAL-na) [reply src/application/services/useAppState.ts:1]: In one of comments you said that in user section of useAppState we name props without "user", because it is already related to user, so I changed naming to `editorTool`

Comment (neSpecc) [reply src/application/services/useAppState.ts:1]: Yes, but it doesn't related to marketplace. The feature you're implementing does not require these changes. So they are just a mess for current PR 

If you really want to change that name just to improve naming, it's better to do it via separate PR

Comment (MedVAL-na) [reply src/application/services/useAppState.ts:1]: Okay, returned naming back

File: src/application/services/useEditorTool.ts (hunk 7 above)
Comment (MedVAL-na) [reply src/application/services/useEditorTool.ts]: moved to onMounted

File: src/application/services/useEditorTool.ts (hunk 13 above)
Comment (MedVAL-na) [reply src/application/services/useEditorTool.ts]: renamed

File: src/presentation/components/editor/Editor.vue
Comment (MedVAL-na) [reply src/presentation/components/editor/Editor.vue:1]: changed naming back

**Review by neSpecc — APPROVED** [review]

**Review by shvkhzod — APPROVED** [review]
